### PR TITLE
Model filters: extract filters from ModelVisitor

### PR DESCRIPTION
**Pull request by Morriar**
A list of filters that can be applied on a MEntity

By default ModelFilter accepts all mentity.

~~~nit
var filter = new ModelFilter
assert filter.accept_mentity(my_mentity) == true
~~~

To quickly configure the filters, options can be passed to the constructor:
~~~nit
var filter = new ModelFilter(
        min_visibility = protected_visibility,
	accept_fictive = false,
	accept_test = false,
	accept_redef = false,
	accept_extern = false,
	accept_attribute = false,
	accept_empty_doc = false
)
~~~

With this, one can use temporary filters with the model visitors and views:

~~~nit
var default_filter = new ModelFilter(private_visibility)
var view = new ModelView(view, default_filter)
# ...
if view.accept_mentity(mentity) then
   # ...
end
# ...
var custom_filter = new ModelFilter(public_visibility)
if view.accept_mentity(mentity, custom_filter) then
   # ...
end
~~~

### CODE REVIEW

File: src/web/api_graph.nit [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# This file is part of NIT ( http://www.nitlanguage.org ).
+
```
Comment (privat) [inline src/web/api_graph.nit]: why?

Comment (Morriar) [reply src/web/api_graph.nit]: The file needed some time alone.

**Comment by Morriar** [comment]
reroll: fixed the removed copyright line

**Comment by privat** [comment]
http://gresil.org/jenkins/job/CI_github/5498/testReport/junit/cmd/nit/errors/

**Pull request merged**